catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/utils/DataSourceUtils.java

```diff
@@ -51,7 +51,7 @@ private static DataSource createDBCPDataSource(JdbcConfig jdbcConfig) throws Exc
     basicDataSource.setMinIdle(jdbcConfig.getPoolMinSize());
     // Set each time a connection is taken out from the connection pool, a test statement will be
     // executed to confirm whether the connection is valid.
-    basicDataSource.setTestOnBorrow(true);
+    basicDataSource.setTestOnBorrow(jdbcConfig.getTestOnBorrow());
```

Comment:
The decision to set testOnBorrow to true or false in a production environment ultimately depends on your specific application requirements and the environment in which it operates. Here are a few considerations to guide you:

- If reliability is a paramount concern and your application cannot afford to encounter errors stemming from invalid database connections (e.g., in financial transactions, critical infrastructure management), it might be wise to set testOnBorrow to true. This ensures every connection from the pool is validated before use, minimizing the chances of connection-related errors at the cost of some performance overhead.

- If performance is critically important and your application operates under high load, where every millisecond of response time counts (e.g., high-frequency trading platforms, real-time analytics), you might opt to set testOnBorrow to false to avoid the extra latency introduced by connection validation. In such cases, it's crucial to have other strategies in place for quickly detecting and recovering from database connectivity issues, such as robust error handling and connection retry mechanisms.

For a balanced approach, consider the following:

- Enable testOnBorrow but use a highly efficient validation query to minimize the performance impact.
Utilize additional settings like testWhileIdle and testOnReturn to keep the connection pool healthy without validating every single borrow action.
- Regularly monitor your application and database performance to assess if the configuration remains optimal under changing load conditions and usage patterns.
- Given these considerations, start with testOnBorrow enabled if you are unsure or if you have not yet had the opportunity to thoroughly test your application's performance and error rates under realistic conditions. This default stance errs on the side of caution, prioritizing reliability. As you gather more data and insights into how your application behaves in production, you can make a more informed decision to either keep it enabled, potentially tweaking the validation query for efficiency, or disable it if the performance cost outweighs the benefits in terms of reliability.